backend/copr_backend/actions.py

```diff
@@ -87,12 +87,19 @@ def __init__(self, opts, action, log=None):
         self.opts = opts
         self.data = action
 
+        self.ext_data = json.loads(action.get("data", "{}"))
+
         self.destdir = self.opts.destdir
         self.front_url = self.opts.frontend_base_url
         self.results_root_url = self.opts.results_baseurl
 
         self.log = log if log else get_redis_logger(self.opts, "backend.actions", "actions")
 
+        storage_enum = StorageEnum.backend
+        if isinstance(self.ext_data, dict) and "storage" in self.ext_data:
+            storage_enum = self.ext_data.get("storage")
```

Comment:
Ah, we are on backend - not frontend :-( so we do not have the models.Action object in hand...